### DISCUSSION TITLE
DIN Rail holde rfor Teensy 4.1 board

### DISCUSSION THREAD

**Discussion by motimtim** [phil-barrett/grblHAL-teensy-4.x#82] [Ideas]
I made a 3D printed DIN rail holder for the Teensy 4.1 board that I am using in a project and was going to post it to the github site, but uploads are disabled and, I have no clue how to use Git. Also, this does not allow me to post an STL file in this discussion. Any idea how I can get to to you? Are you interested in it?
Thanks,
Tim

![PXL_20230925_124755027 MP](https://github.com/phil-barrett/grblHAL-teensy-4.x/assets/57264756/9deacc94-2ace-468f-9393-7f1b75ae4e41)
![PXL_20230925_124802385](https://github.com/phil-barrett/grblHAL-teensy-4.x/assets/57264756/628d17a0-0ff6-42bd-8fdb-a746b33dd797)

**Comment by motimtim** [comment]
My apologies. I did not see the post several spots down. I hope this can still be of help.

**Reply by phil-barrett** [reply]
Excellent! I love DIN rail stuff. You should be able to attach a zip file.  I can move it into the repo and give you full credit.  Did you use PLA or PETG?

By the way, you might want to use 4-40 or M3 screws for better clearance with the screw terminals. The ones you are using look a bit big.  
I have a number of questions but will hold off until I see the STL (through hole clearance and locking/release for the DIN rail.

**Comment by motimtim** [comment]
I used 3/8" long #4-20 Plastite screws. Hole size should be fine for tapped M3 or 4-40 as well.

I printed in PLA with a fine layer height with a 0.4mm nozzle. This type of clip for me has worked on a few printers for me, so it should work pretty well. Locking is robust, unlocking works, but is not perfect but still more than adequate.

[GRBL Board - T4.1 DIN Rail Holder.STL.zip](https://github.com/phil-barrett/grblHAL-teensy-4.x/files/12718175/GRBL.Board.-.T4.1.DIN.Rail.Holder.STL.zip)

**Comment by motimtim** [comment]
FYI, I am using this on a fully populated board now and it is working well. Screws fit without issue as well.
